### DISCUSSION TITLE
`remotelogger.New` should return a pointer instead of value

### DISCUSSION THREAD

**Discussion by maohieng** [gofr-dev/gofr#653] [Q&A] [answered]
I'm curious why the function `remotelogger.New` returns a value instead of a pointer, especially since it is treated as a pointer throughout the code. Am I missing something? Please advice.
```golang
func New(level logging.Level, remoteConfigURL, loggerFetchInterval string) logging.Logger {
	interval, err := strconv.Atoi(loggerFetchInterval)
	if err != nil {
		interval = 15
	}

	l := remoteLogger{
		remoteURL:          remoteConfigURL,
		Logger:             logging.NewLogger(level),
		levelFetchInterval: interval,
		currentLevel:       level,
	}

	if remoteConfigURL != "" {
		go l.UpdateLogLevel()
	}

	return l
}
```

Should it be 
```golang
	l := &remoteLogger{
		remoteURL:          remoteConfigURL,
		Logger:             logging.NewLogger(level),
		levelFetchInterval: interval,
		currentLevel:       level,
	}
```
?

**Comment by vipul-rawat** [comment, marked as answer]
Yes, we can to keep it uniform, but since we do not have methods defined to override the Logger interface(return type) we may not need to change the return to a pointer. Changing to a pointer does not seem to give us any benefit as when the interface is passed around since the underlying type is a large struct it is by design passed as a pointer.
